plugins/permission-react/src/components/PermissionedRoute.tsx

```diff
@@ -28,13 +32,25 @@ import { Permission } from '@backstage/plugin-permission-common';
  */
 export const PermissionedRoute = (
   props: ComponentProps<typeof Route> & {
-    permission: Permission;
-    resourceRef?: string;
     errorComponent?: ReactElement | null;
-  },
+  } & (
+      | {
+          permission: Exclude<Permission, ResourcePermission>;
+          resourceRef?: never;
+        }
+      | {
+          permission: ResourcePermission;
+          resourceRef: string | undefined;
+        }
+    ),
 ) => {
   const { permission, resourceRef, errorComponent, ...otherProps } = props;
-  const permissionResult = usePermission(permission, resourceRef);
+
+  const permissionResult = usePermission(
+    isResourcePermission(permission)
+      ? { permission, resourceRef }
+      : { permission },
```

Comment:
Guess this is a case for making the `isResourcePermission` make sure the `resourceRef` isn't only present but also truthy/non-empty string?

Reply:
Perhaps I'm not fully understanding, but to me `isResourcePermission` shouldn't do anything with the `resourceRef`, since it's not part of the permission.

Reply:
Was thinking about using the shorthand of `usePermission({ permission, resourceRef })` here, and just let `resourceRef` be undefined. Realizing the `in` check is at another level and this is intentionally designed this way to catch mistakes, so nvm!